Apps/Database/src/Delegates/DBCommunicationDelegate.cs

```diff
@@ -146,5 +146,33 @@ public List<Communication> GetEmailCommunicationsToSend()
             this.logger.LogDebug($"Finished getting list of New & Processing Email Communications from DB. {JsonSerializer.Serialize(retVal)}");
             return retVal;
         }
+
+        /// <inheritdoc />
+        public DBResult<Communication> Delete(Communication communication, bool commit = true)
+        {
+            this.logger.LogTrace($"Deleting Communication from DB...");
+            DBResult<Communication> result = new DBResult<Communication>()
+            {
+                Payload = communication,
+                Status = DBStatusCode.Deferred,
+            };
+            this.dbContext.Communication.Remove(communication);
+            if (commit)
+            {
+                try
+                {
+                    this.dbContext.SaveChanges();
+                    result.Status = DBStatusCode.Deleted;
+                }
+                catch (DbUpdateConcurrencyException e)
```

Comment:
I would probably change this to DbUpdateException to catch anything wrong with the delete.

Reply:
updated to DbUpdateException